File: apps/android/app/src/main/java/com/thiratt/nekoshare/features/home/model/Friend.kt
package com.thiratt.nekoshare.features.home.model

enum class FriendStatus {
    Friend,
    Incoming,
    Outgoing,
    Blocked
}

data class Friend(
    val id: String,
    val name: String,
    val username: String,
    val avatarUrl: String? = null,
    val status: FriendStatus,
    val isOnline: Boolean = false,
    val lastSeen: String = ""
)

File: packages/contracts/src/api/friend.ts
export type FriendStatus = "none" | "outgoing" | "incoming" | "friend" | "blocked";

export interface FriendItem {
	id: string;
	friendId: string;
	name: string;
	email: string;
	avatarUrl?: string;
	status: FriendStatus;
	lastActive: string;
	createdAt: string;
	isOnline: boolean;
}

export interface FriendListResponse {
	friends: FriendItem[];
	incoming: FriendItem[];
	outgoing: FriendItem[];
	total: { friends: number; incoming: number; outgoing: number };
}

export interface FriendRequestPayload {
	userId: string;
}

export interface FriendActionResponse {
	success: boolean;
	friendId: string;
	status: FriendStatus;
}

export interface UserSearchResult {
	id: string;
	name: string;
	email: string;
	avatarUrl?: string;
	friendStatus: FriendStatus;
}

export interface UserSearchResponse {
	users: UserSearchResult[];
	total: number;
}
